File: ruby/raindrops/raindrops.rb
class Raindrops
  VERSION = 1
  
  def self.convert(num)
    drops = ""
    if num%105==0 # 3,5,7
      return "PlingPlangPlong"
    elsif num %15==0 #3,5
      return "PlingPlang"
    elsif num%21==0 #3,7
      return "PlingPlong"
    elsif num%35==0 #5,7
      return "PlangPlong"
    elsif num %3==0
      return "Pling"
    elsif num % 5 == 0
      return 'Plang'
    elsif num % 7 == 0
      return 'Plong'
    else
      return num.to_s
    end
  end
end

File: go/raindrops/raindrops.go
package raindrops

import "strconv"

const testVersion = 2

// Convert returns a string of combos of Pling, Plang and Plong depending on the number's primes
func Convert(incomingNumber int) string {
	raindrops := ""

	if incomingNumber%3 == 0 {
		raindrops = raindrops + "Pling"
	}
	if incomingNumber%5 == 0 {
		raindrops = raindrops + "Plang"
	}
	if incomingNumber%7 == 0 {
		raindrops = raindrops + "Plong"
	}
	if raindrops == "" {
		raindrops = strconv.Itoa(incomingNumber)
	}

	return raindrops
}

// The test program has a benchmark too.  How fast does your Convert convert?
